src/core/public/http/fetch.ts

```diff
@@ -95,38 +102,53 @@ export class Fetch {
     });
   };
 
-  private createRequest(path: string, options?: HttpFetchOptions): Request {
+  private createRequest(options: HttpFetchOptionsWithPath): Request {
     // Merge and destructure options out that are not applicable to the Fetch API.
-    const { query, prependBasePath: shouldPrependBasePath, asResponse, ...fetchOptions } = merge(
+    const {
+      query,
+      prependBasePath: shouldPrependBasePath,
+      asResponse,
+      asSystemRequest,
+      ...fetchOptions
+    } = merge(
       {
         method: 'GET',
         credentials: 'same-origin',
         prependBasePath: true,
+      },
+      options,
+      {
         headers: {
-          'kbn-version': this.params.kibanaVersion,
           'Content-Type': 'application/json',
+          ...options.headers,
+          'kbn-version': this.params.kibanaVersion,
         },
-      },
-      options || {}
+      }
     );
     const url = format({
-      pathname: shouldPrependBasePath ? this.params.basePath.prepend(path) : path,
+      pathname: shouldPrependBasePath ? this.params.basePath.prepend(options.path) : options.path,
       query,
     });
 
-    if (
-      options &&
-      options.headers &&
-      'Content-Type' in options.headers &&
-      options.headers['Content-Type'] === undefined
-    ) {
-      delete fetchOptions.headers['Content-Type'];
+    // Make sure the system request header is only present if `asSystemRequest` is true.
+    if (asSystemRequest) {
+      fetchOptions.headers['kbn-system-request'] = 'true';
+    } else {
+      if ('kbn-system-request' in fetchOptions.headers) {
+        Reflect.deleteProperty(fetchOptions.headers, 'kbn-system-request');
+      }
+    }
+
+    // Deprecated header used by legacy platform pre-7.7. Remove in 8.x.
+    if ('kbn-system-api' in fetchOptions.headers) {
+      Reflect.deleteProperty(fetchOptions.headers, 'kbn-system-api');
```

Comment:
TIL this exists. So I guess we prefer this to an old-school `delete`?

Reply:
I don't think we have a preference or convention. I just find this a bit more explicit personally. `delete` is a weird statement IMO. Also it can't be mocked out if you wanted to for some reason.